pkg/operator/observe_config.go

```diff
@@ -97,6 +101,58 @@ func (c ConfigObserver) sync() error {
 	return nil
 }
 
+// observeClusterConfig observes cloud provider configuration from
+// cluster-config-v1 in order to configure kube-controller-manager's cloud
+// provider.
+func observeClusterConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	clusterConfig, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		glog.Warningf("cluster-config-v1 not found in the kube-system namespace")
+		return observedConfig, nil
+	}
+	if err != nil {
+		return observedConfig, err
+	}
+
+	installConfigYaml, ok := clusterConfig.Data["install-config"]
+	if !ok {
+		return observedConfig, nil
+	}
+	installConfig := map[string]interface{}{}
```

Comment:
would like to see if we can vendor installer's type rather than this map[string]interface{}, install-config has Name func on platform [see here](https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L74-L91) that can be used to make decisions.

Reply:
> would like to see if we can vendor installer's type rather than this map[string]interface{}, install-config has Name func on platform [see here](https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L74-L91) that can be used to make decisions.

Once the type is promoted to `openshift/api` we can do that, but I don't want to start having different operators depend on each other for types.  If you do that, we'll end up wedged as we try to rebase to various levels of kube.

Reply:
Yeah, this is a pervasive issue that cuts across all my team's operators (and now this one). Not going to solve it with this PR.